File: src/main/java/com/heekng/celloct/entity/JoinRequest.java
package com.heekng.celloct.entity;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import static javax.persistence.FetchType.*;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(
        uniqueConstraints = {
                @UniqueConstraint(
                        name = "join_request_member_id_shop_id_unique",
                        columnNames = {"member_id", "shop_id"}
                )
        }
)
public class JoinRequest extends BaseTimeEntity{

    @Id @GeneratedValue
    @Column(name = "join_request_id")
    private Long id;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "shop_id", nullable = false)
    private Shop shop;

    @Builder
    public JoinRequest(Member member, Shop shop) {
        this.member = member;
        this.shop = shop;
    }
}

File: src/main/kotlin/com/heekng/celloct/entity/JoinRequest.kt
package com.heekng.celloct.entity

import javax.persistence.*

@Entity
@Table(
    uniqueConstraints = [UniqueConstraint(
        name = "join_request_member_id_shop_id_unique",
        columnNames = ["member_id", "shop_id"]
    )]
)
class JoinRequest(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) @Column(name = "join_request_id")
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "member_id", nullable = false)
    val member: Member,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "shop_id", nullable = false)
    val shop: Shop,
) : BaseTimeEntity() {

    companion object {
        fun fixture(
            member: Member,
            shop: Shop,
        ): JoinRequest {
            return JoinRequest(
                member = member,
                shop = shop
            )
        }
    }

}
